core/src/main/java/com/graphhopper/reader/osm/OSMReader.java

```diff
@@ -195,17 +195,16 @@ protected boolean isBarrierNode(ReaderNode node) {
      * This method is called during the second pass of {@link WaySegmentParser} and provides an entry point to enrich
      * the given OSM way with additional tags before it is passed on to the tag parsers.
      */
-    protected void setArtificialWayTags(PointList pointList, ReaderWay way) {
-        // Estimate length of ways containing a route tag e.g. for ferry speed calculation
+    protected void setArtificialWayTags(PointList pointList, ReaderWay way, double distance) {
         double firstLat = pointList.getLat(0), firstLon = pointList.getLon(0);
         double lastLat = pointList.getLat(pointList.size() - 1), lastLon = pointList.getLon(pointList.size() - 1);
+        way.setTag("beeline_distance", distCalc.calcDist(firstLat, firstLon, lastLat, lastLon));
+        way.setTag("road_distance", distance);
```

Comment:
Or should we call it `road_length`?

Reply:
road_distance is fine for me (or edge_distance?)

Do we need the separate beeline_distance only for curvature? Then we could move the calculation there and avoid it here.

Reply:
First I thought `edge_distance` is not a good choice, because it sounded like the distance/length of the (straight line 'edge'/connection between the tower nodes). But actually it's consistent to call it like this, because it is equal to `edge.getDistance()`.

> Do we need the separate beeline_distance only for curvature? Then we could move the calculation there and avoid it here.

:+1: